File: src/main/java/com/d20charactersheet/framework/dac/dao/sql/rowmapper/SkillRowMapper.java
package com.d20charactersheet.framework.dac.dao.sql.rowmapper;

import org.jetbrains.annotations.NotNull;

import com.d20charactersheet.framework.boc.model.Attribute;
import com.d20charactersheet.framework.boc.model.Skill;
import com.d20charactersheet.framework.dac.dao.BaseRowMapper;
import com.d20charactersheet.framework.dac.dao.DataRow;

/**
 * Maps character independent skill data to a skill.
 */
public class SkillRowMapper extends BaseRowMapper {

    /**
     * Maps character independent skill data to a skill.
     */
    @NotNull
    @Override
    public Object mapRow(@NotNull DataRow dataRow) throws java.sql.SQLException {
        final Skill skill = new Skill();
        skill.setId(dataRow.getInt(0));
        skill.setAttribute((Attribute) getEnum(dataRow.getInt(1), Attribute.values()));
        skill.setUntrained(getBoolean(dataRow.getInt(2)));
        skill.setName(dataRow.getString(3));
        return skill;
    }
}

File: src/main/java/com/d20charactersheet/framework/dac/dao/sql/rowmapper/SkillRowMapper.kt
package com.d20charactersheet.framework.dac.dao.sql.rowmapper

import com.d20charactersheet.framework.boc.model.Attribute
import com.d20charactersheet.framework.boc.model.Skill
import com.d20charactersheet.framework.dac.dao.sql.DataRow
import java.sql.SQLException

/**
 * Maps character independent skill data to a skill.
 */
class SkillRowMapper : BaseRowMapper() {

    /**
     * Maps character independent skill data to a skill.
     */
    @Throws(SQLException::class)
    override fun mapRow(dataRow: DataRow): Any {
        val skill = Skill()
        skill.id = dataRow.getInt(0)
        skill.attribute = Attribute.values()[dataRow.getInt(1)]
        skill.isUntrained = getBoolean(dataRow.getInt(2))
        skill.name = dataRow.getString(3)
        return skill
    }
}